base/src/commonMain/kotlin/flow/SequentialMutableSharedFlow.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 Splendo Consulting B.V. The Netherlands
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.splendo.kaluga.base.flow
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.flow.FlowCollector
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class SequentialMutableSharedFlow<T>(replay: Int = 0, extraBufferCapacity: Int = 0, coroutineScope: CoroutineScope) : MutableSharedFlow<T>, CoroutineScope by coroutineScope {
```

Comment:
Some KDoc would be nice for this one. I know why this exists but others might not.

Reply:
Also I would follow pattern that SequentialMutableSharedFlow is an `interface`

Reply:
or at least an interface just for `tryEmitOrLaunchEmit`